protocol/state/outputs.go

```diff
@@ -27,19 +30,23 @@ func NewOutput(o bc.TxOutput, p bc.Outpoint) *Output {
 // excludes reference data).
 func Prevout(in *bc.TxInput) *Output {
 	assetAmount := in.AssetAmount()
+	// TODO(oleg): for new outputid we need to have correct output commitment, not reconstruct this here
+	// Also we do not care about all these, but only about UnspentID
 	t := bc.NewTxOutput(assetAmount.AssetID, assetAmount.Amount, in.ControlProgram(), nil)
 	return &Output{
-		Outpoint: in.Outpoint(),
+		OutputID: in.SpentOutputID(),
 		TxOutput: *t,
 	}
 }
 
 // OutputKey returns the key of an output in the state tree.
-func OutputKey(o bc.Outpoint) (bkey []byte) {
-	var b bytes.Buffer
-	w := errors.NewWriter(&b) // used to satisfy interfaces
-	o.WriteTo(w)
-	return b.Bytes()
+func OutputKey(o bc.UnspentID) (bkey []byte) {
+	// TODO(oleg): check if we no longer need this buffer writing.
```

Comment:
You're right, we don't need it. The buffer writing was just to serialize the outpoint. Since an UnspentID's internal representation is the actual bytes we want to put in the tree, we can just return them as you're doing here.